server/src/main/java/org/spine3/server/storage/memory/InMemoryRecordStorage.java

```diff
@@ -0,0 +1,173 @@
+                                                         .build();
+
+    private final Map<TenantId, Map<I, EntityStorageRecord>> tenantToStorageMap = newHashMap();
+
+    protected InMemoryRecordStorage(boolean multitenant) {
+        super(multitenant);
+    }
+
+    @SuppressWarnings("MethodWithMultipleLoops")    /* It's OK for in-memory implementation
+                                                     * as it is used primarily in tests. */
+    @Override
+    protected Iterable<EntityStorageRecord> readBulkInternal(final Iterable<I> givenIds, FieldMask fieldMask) {
+        final Map<I, EntityStorageRecord> storage = getStorage();
+
+        // It is not possible to return an immutable collection, since {@code null} may be present in it.
+        final Collection<EntityStorageRecord> result = new LinkedList<>();
+
+        TypeUrl typeUrl = null;
+
+        for (I recordId : storage.keySet()) {
+            for (I givenId : givenIds) {
+                if (recordId.equals(givenId)) {
+                    final EntityStorageRecord.Builder matchingRecord = storage.get(recordId).toBuilder();
+                    final Any state = matchingRecord.getState();
+
+                    if (typeUrl == null) {
+                        typeUrl = TypeUrl.of(state.getTypeUrl());
+                    }
+
+                    final Message wholeState = AnyPacker.unpack(state);
+                    final Message maskedState = FieldMasks.applyIfValid(fieldMask, wholeState, typeUrl);
+                    final Any processed = AnyPacker.pack(maskedState);
+
+                    matchingRecord.setState(processed);
+
+                    result.add(matchingRecord.build());
+                } else {
+                    result.add(null);
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    protected Iterable<EntityStorageRecord> readBulkInternal(Iterable<I> ids) {
+        return readBulkInternal(ids, null);
+    }
+
+    @Override
+    protected Map<I, EntityStorageRecord> readAllInternal() {
+        final Map<I, EntityStorageRecord> storage = getStorage();
+
+        final ImmutableMap<I, EntityStorageRecord> result = ImmutableMap.copyOf(storage);
+        return result;
+    }
+
+    @Override
+    protected Map<I, EntityStorageRecord> readAllInternal(FieldMask fieldMask) {
+        if (fieldMask.getPathsList().isEmpty()) {
+            return readAllInternal();
+        }
+
+        final Map<I, EntityStorageRecord> storage = getStorage();
+
+        if (storage.isEmpty()) {
+            return newHashMap();
+        }
+
+        final ImmutableMap.Builder<I, EntityStorageRecord> result = ImmutableMap.builder();
+
+        final Collection<Message> records = FieldMasks.applyMask(fieldMask, Collections2.transform(storage.values(), new Function<EntityStorageRecord, Message>() {
+            @Nullable
+            @Override
+            public Message apply(@Nullable EntityStorageRecord input) {
```

Comment:
This function looks like a reusable piece of code. Don't we use such a thing elsewhere?